staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -1453,21 +1457,12 @@ func RunTestGetListNonRecursive(ctx context.Context, t *testing.T, store storage
 			},
 		},
 		expectedOut: []example.Pod{},
-		rv:          fmt.Sprintf("%d", currentRV),
+		rv:          fmt.Sprintf("%d", objRV),
 	}}
 
 	for _, tt := range tests {
 		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
-			// For some asynchronous implementations of storage interface (in particular watchcache),
-			// certain requests may impact result of further requests. As an example, if we first
-			// ensure that watchcache is synchronized up to ResourceVersion X (using Get/List requests
-			// with NotOlderThan semantic), the further requests (even specifying earlier resource
-			// version) will also return the result synchronized to at least ResourceVersion X.
-			// By parallelizing test cases we ensure that the order in which test cases are defined
-			// doesn't automatically preclude some scenarios from happening.
-			t.Parallel()
```

Comment:
@wojtek-t Removed this as pararelization breaks setting a global feature flag. Any suggestions how we can get restore the previous behavior.

One idea would be to avoid calling t.Parallel on the last test case, so only test cases from RunTestGetListNonRecursive are parallelized.

Reply:
Filled https://github.com/kubernetes/kubernetes/issues/123677

Reply:
I really don't want to do that - the parallelization here allowed us to discover some bugs in the past and I don't want to give up on it.

IIRC, t.Parallel() is effective within a single test file, so I think a mitigation would be to have the test that set this feature flag in separate files?

Reply:
Separating files is not enough :(, need separate modules.

Reply:
Separate modules is safe, but mixing serial and parallel tests inside a single package should also be possible. Parallel tests only run concurrently to other tests which are also marked as parallel.

In this example, TestA and TestB run sequentially, with sub-tests of TestB running in parallel:
```
func TestA(t *testing.T) { ... }

func TestB(t *testing.T) {
   ...
   for _, tc := range testcases {
      tc := tc
      t.Run(tc.Name, func(t *testing.T) {
          t.Parallel()
          ...
      })
  }
}
```